extensions/resource-deployment/src/ui/wizardBase.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { Disposable } from 'vscode';
+import { WizardPageBase } from './wizardPageBase';
+import { Model } from './model';
+
+export abstract class WizardBase<T, M extends Model> {
+	private customButtons: azdata.window.Button[] = [];
+	private pages: WizardPageBase<T>[] = [];
+
+	public wizardObject: azdata.window.Wizard;
+	public toDispose: Disposable[] = [];
+	public get model(): M {
+		return this._model;
+	}
+
+	constructor(title: string, private _model: M) {
+		this.wizardObject = azdata.window.createWizard(title);
+	}
+
+	public open(): Thenable<void> {
+		this.initialize();
```

Comment:
Why does this need to be Thenable?

Reply:
Thenable is the return type of an API that we are calling inside it.